### PR TITLE
ReturnToPrevious: Check the state of the RTP feature toggle in the hook

### PR DESCRIPTION
**Pull request by eledobleefe**
**What is this feature?**

We add a check for the RTP feature toggle state in the hook.

**Why do we need this feature?**

To automatically do the check, so devs don't need to worry about it when implementing the RTP functionality.

**Who is this feature for?**

Everyone using the RTP functionality.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by ashharrison90 — COMMENTED** [review]
question

File: public/app/core/components/AppChrome/AppChromeService.tsx [hunk 1]
```diff
@@ -91,13 +91,18 @@ export class AppChromeService {
...
+        page: returnToPrevious.href,
+        previousPage: previousPage?.href,
+      });
 
-    this.update({ returnToPrevious });
-    window.sessionStorage.setItem('returnToPrevious', JSON.stringify(returnToPrevious));
+      this.update({ returnToPrevious });
+      window.sessionStorage.setItem('returnToPrevious', JSON.stringify(returnToPrevious));
+    } else {
+      locationService.push(returnToPrevious.href);
```
[the remark this answers is not in the document]
Comment (eledobleefe) [reply public/app/core/components/AppChrome/AppChromeService.tsx:104]: 😲 Let me double-check!

[the remark this answers is not in the document]
Comment (joshhunt) [reply public/app/core/components/AppChrome/AppChromeService.tsx:104]: right - if the toggle is off, the hook just becomes a no-op.

**Review by ashharrison90 — APPROVED** [review]
lgtm 👍

File: public/app/core/components/AppChrome/AppChromeService.tsx [hunk 2]
```diff
@@ -91,6 +91,10 @@ export class AppChromeService {
 
   public setReturnToPrevious = (returnToPrevious: ReturnToPreviousProps) => {
     const previousPage = this.state.getValue().returnToPrevious;
+    const isReturnToPreviousEnabled = config.featureToggles.returnToPrevious;
+    if (!isReturnToPreviousEnabled) {
+      return;
+    }
```
Comment (joshhunt) [inline public/app/core/components/AppChrome/AppChromeService.tsx:97]: I think we should:

 - move this up one line (no need to get `previousPage` only to bail out if the toggle isnt set)
 - also add the same check to `clearReturnToPrevious` for thoroughness

**Review by joshhunt — APPROVED** [review]
LETS GOOOOO!!!

**Pull request merged**